Reverts r65963
Use stub executables generated by RubyGems istead of original executables.
diff --git a/bin/bundle b/bin/bundle
index 59dad21c60..aaf773745d 100755
--- a/bin/bundle
+++ b/bin/bundle
@@ -1,27 +1,31 @@
 #!/usr/bin/env ruby
-#
-# This file was generated by RubyGems.
-#
-# The application 'bundler' is installed as part of a gem, and
-# this file is here to facilitate running it.
-#
+# frozen_string_literal: true
 
-require 'rubygems'
-
-version = ">= 0.a"
+# Exit cleanly from an early interrupt
+Signal.trap("INT") do
+  Bundler.ui.debug("\n#{caller.join("\n")}") if defined?(Bundler)
+  exit 1
+end
 
-if ARGV.first
-  str = ARGV.first
-  str = str.dup.force_encoding("BINARY") if str.respond_to? :force_encoding
-  if str =~ /\A_(.*)_\z/ and Gem::Version.correct?($1) then
-    version = $1
-    ARGV.shift
-  end
+require "bundler"
+# Check if an older version of bundler is installed
+$LOAD_PATH.each do |path|
+  next unless path =~ %r{/bundler-0\.(\d+)} && $1.to_i < 9
+  err = String.new
+  err << "Looks like you have a version of bundler that's older than 0.9.\n"
+  err << "Please remove your old versions.\n"
+  err << "An easy way to do this is by running `gem cleanup bundler`."
+  abort(err)
 end
 
-if Gem.respond_to?(:activate_bin_path)
-load Gem.activate_bin_path('bundler', 'bundle', version)
-else
-gem "bundler", version
-load Gem.bin_path("bundler", "bundle", version)
+require "bundler/friendly_errors"
+Bundler.with_friendly_errors do
+  require "bundler/cli"
+
+  # Allow any command to use --help flag to show help for that command
+  help_flags = %w[--help -h]
+  help_flag_used = ARGV.any? {|a| help_flags.include? a }
+  args = help_flag_used ? Bundler::CLI.reformatted_help_args(ARGV) : ARGV
+
+  Bundler::CLI.start(args, :debug => true)
 end
diff --git a/bin/bundle_ruby b/bin/bundle_ruby
index be7fb5afe4..2209c6195f 100755
--- a/bin/bundle_ruby
+++ b/bin/bundle_ruby
@@ -1,27 +1,60 @@
 #!/usr/bin/env ruby
-#
-# This file was generated by RubyGems.
-#
-# The application 'bundler' is installed as part of a gem, and
-# this file is here to facilitate running it.
-#
-
-require 'rubygems'
-
-version = ">= 0.a"
-
-if ARGV.first
-  str = ARGV.first
-  str = str.dup.force_encoding("BINARY") if str.respond_to? :force_encoding
-  if str =~ /\A_(.*)_\z/ and Gem::Version.correct?($1) then
-    version = $1
-    ARGV.shift
+# frozen_string_literal: true
+
+require "bundler/shared_helpers"
+
+Bundler::SharedHelpers.major_deprecation(3, "the bundle_ruby executable has been removed in favor of `bundle platform --ruby`")
+
+Signal.trap("INT") { exit 1 }
+
+require "bundler/errors"
+require "bundler/ruby_version"
+require "bundler/ruby_dsl"
+
+module Bundler
+  class Dsl
+    include RubyDsl
+
+    attr_accessor :ruby_version
+
+    def initialize
+      @ruby_version = nil
+    end
+
+    def eval_gemfile(gemfile, contents = nil)
+      contents ||= File.open(gemfile, "rb", &:read)
+      instance_eval(contents, gemfile.to_s, 1)
+    rescue SyntaxError => e
+      bt = e.message.split("\n")[1..-1]
+      raise GemfileError, ["Gemfile syntax error:", *bt].join("\n")
+    rescue ScriptError, RegexpError, NameError, ArgumentError => e
+      e.backtrace[0] = "#{e.backtrace[0]}: #{e.message} (#{e.class})"
+      STDERR.puts e.backtrace.join("\n       ")
+      raise GemfileError, "There was an error in your Gemfile," \
+        " and Bundler cannot continue."
+    end
+
+    def source(source, options = {})
+    end
+
+    def gem(name, *args)
+    end
+
+    def group(*args)
+    end
   end
 end
 
-if Gem.respond_to?(:activate_bin_path)
-load Gem.activate_bin_path('bundler', 'bundle_ruby', version)
-else
-gem "bundler", version
-load Gem.bin_path("bundler", "bundle_ruby", version)
+dsl = Bundler::Dsl.new
+begin
+  dsl.eval_gemfile(Bundler::SharedHelpers.default_gemfile)
+  ruby_version = dsl.ruby_version
+  if ruby_version
+    puts ruby_version
+  else
+    puts "No ruby version specified"
+  end
+rescue Bundler::GemfileError => e
+  puts e.message
+  exit(-1)
 end
diff --git a/bin/bundler b/bin/bundler
index 7ef736899d..d9131fe834 100755
--- a/bin/bundler
+++ b/bin/bundler
@@ -1,27 +1,4 @@
 #!/usr/bin/env ruby
-#
-# This file was generated by RubyGems.
-#
-# The application 'bundler' is installed as part of a gem, and
-# this file is here to facilitate running it.
-#
+# frozen_string_literal: true
 
-require 'rubygems'
-
-version = ">= 0.a"
-
-if ARGV.first
-  str = ARGV.first
-  str = str.dup.force_encoding("BINARY") if str.respond_to? :force_encoding
-  if str =~ /\A_(.*)_\z/ and Gem::Version.correct?($1) then
-    version = $1
-    ARGV.shift
-  end
-end
-
-if Gem.respond_to?(:activate_bin_path)
-load Gem.activate_bin_path('bundler', 'bundler', version)
-else
-gem "bundler", version
-load Gem.bin_path("bundler", "bundler", version)
-end
+load File.expand_path("../bundle", __FILE__)
diff --git a/bin/irb b/bin/irb
index e44609a1b0..c64ee85fbd 100755
--- a/bin/irb
+++ b/bin/irb
@@ -1,27 +1,11 @@
 #!/usr/bin/env ruby
 #
-# This file was generated by RubyGems.
+#   irb.rb - interactive ruby
+#   	$Release Version: 0.9.6 $
+#   	$Revision$
+#   	by Keiju ISHITSUKA(keiju@ruby-lang.org)
 #
-# The application 'irb' is installed as part of a gem, and
-# this file is here to facilitate running it.
-#
-
-require 'rubygems'
-
-version = ">= 0.a"
 
-if ARGV.first
-  str = ARGV.first
-  str = str.dup.force_encoding("BINARY") if str.respond_to? :force_encoding
-  if str =~ /\A_(.*)_\z/ and Gem::Version.correct?($1) then
-    version = $1
-    ARGV.shift
-  end
-end
+require "irb"
 
-if Gem.respond_to?(:activate_bin_path)
-load Gem.activate_bin_path('irb', 'irb', version)
-else
-gem "irb", version
-load Gem.bin_path("irb", "irb", version)
-end
+IRB.start(__FILE__)
diff --git a/bin/rdoc b/bin/rdoc
index 74e8b025b4..aaa23292df 100755
--- a/bin/rdoc
+++ b/bin/rdoc
@@ -1,27 +1,44 @@
 #!/usr/bin/env ruby
 #
-# This file was generated by RubyGems.
-#
-# The application 'rdoc' is installed as part of a gem, and
-# this file is here to facilitate running it.
+#  RDoc: Documentation tool for source code
+#        (see lib/rdoc/rdoc.rb for more information)
 #
+#  Copyright (c) 2003 Dave Thomas
+#  Released under the same terms as Ruby
 
-require 'rubygems'
+begin
+  gem 'rdoc'
+rescue NameError => e # --disable-gems
+  raise unless e.name == :gem
+rescue Gem::LoadError
+end
 
-version = ">= 0.a"
+require 'rdoc/rdoc'
 
-if ARGV.first
-  str = ARGV.first
-  str = str.dup.force_encoding("BINARY") if str.respond_to? :force_encoding
-  if str =~ /\A_(.*)_\z/ and Gem::Version.correct?($1) then
-    version = $1
-    ARGV.shift
+begin
+  r = RDoc::RDoc.new
+  r.document ARGV
+rescue Errno::ENOSPC
+  $stderr.puts 'Ran out of space creating documentation'
+  $stderr.puts
+  $stderr.puts 'Please free up some space and try again'
+rescue SystemExit
+  raise
+rescue Exception => e
+  if $DEBUG_RDOC then
+    $stderr.puts e.message
+    $stderr.puts "#{e.backtrace.join "\n\t"}"
+    $stderr.puts
+  elsif Interrupt === e then
+    $stderr.puts
+    $stderr.puts 'Interrupted'
+  else
+    $stderr.puts "uh-oh! RDoc had a problem:"
+    $stderr.puts e.message
+    $stderr.puts
+    $stderr.puts "run with --debug for full backtrace"
   end
-end
 
-if Gem.respond_to?(:activate_bin_path)
-load Gem.activate_bin_path('rdoc', 'rdoc', version)
-else
-gem "rdoc", version
-load Gem.bin_path("rdoc", "rdoc", version)
+  exit 1
 end
+
diff --git a/bin/ri b/bin/ri
index eaca302681..7fbed0c099 100755
--- a/bin/ri
+++ b/bin/ri
@@ -1,27 +1,12 @@
 #!/usr/bin/env ruby
-#
-# This file was generated by RubyGems.
-#
-# The application 'rdoc' is installed as part of a gem, and
-# this file is here to facilitate running it.
-#
 
-require 'rubygems'
-
-version = ">= 0.a"
-
-if ARGV.first
-  str = ARGV.first
-  str = str.dup.force_encoding("BINARY") if str.respond_to? :force_encoding
-  if str =~ /\A_(.*)_\z/ and Gem::Version.correct?($1) then
-    version = $1
-    ARGV.shift
-  end
+begin
+  gem 'rdoc'
+rescue NameError => e # --disable-gems
+  raise unless e.name == :gem
+rescue Gem::LoadError
 end
 
-if Gem.respond_to?(:activate_bin_path)
-load Gem.activate_bin_path('rdoc', 'ri', version)
-else
-gem "rdoc", version
-load Gem.bin_path("rdoc", "ri", version)
-end
+require 'rdoc/ri/driver'
+
+RDoc::RI::Driver.run ARGV
diff --git a/common.mk b/common.mk
index f927829166..e75051801c 100644
--- a/common.mk
+++ b/common.mk
@@ -542,15 +542,15 @@ post-install-gem::
 
 rdoc: PHONY main
 	@echo Generating RDoc documentation
-	$(Q) $(XRUBY) "$(srcdir)/libexec/rdoc" --root "$(srcdir)" --page-dir "$(srcdir)/doc" --encoding=UTF-8 --no-force-update --all --ri --op "$(RDOCOUT)" $(RDOCFLAGS) "$(srcdir)"
+	$(Q) $(XRUBY) "$(srcdir)/bin/rdoc" --root "$(srcdir)" --page-dir "$(srcdir)/doc" --encoding=UTF-8 --no-force-update --all --ri --op "$(RDOCOUT)" $(RDOCFLAGS) "$(srcdir)"
 
 html: PHONY main
 	@echo Generating RDoc HTML files
-	$(Q) $(XRUBY) "$(srcdir)/libexec/rdoc" --root "$(srcdir)" --page-dir "$(srcdir)/doc" --encoding=UTF-8 --no-force-update --all --op "$(HTMLOUT)" $(RDOCFLAGS) "$(srcdir)"
+	$(Q) $(XRUBY) "$(srcdir)/bin/rdoc" --root "$(srcdir)" --page-dir "$(srcdir)/doc" --encoding=UTF-8 --no-force-update --all --op "$(HTMLOUT)" $(RDOCFLAGS) "$(srcdir)"
 
 rdoc-coverage: PHONY main
 	@echo Generating RDoc coverage report
-	$(Q) $(XRUBY) "$(srcdir)/libexec/rdoc" --root "$(srcdir)" --encoding=UTF-8 --all --quiet -C $(RDOCFLAGS) "$(srcdir)"
+	$(Q) $(XRUBY) "$(srcdir)/bin/rdoc" --root "$(srcdir)" --encoding=UTF-8 --all --quiet -C $(RDOCFLAGS) "$(srcdir)"
 
 RDOCBENCHOUT=/tmp/rdocbench
 
diff --git a/lib/bundler.gemspec b/lib/bundler.gemspec
index 2b2bb412d5..96b9b7861b 100644
--- a/lib/bundler.gemspec
+++ b/lib/bundler.gemspec
@@ -58,7 +58,7 @@
   # include the gemspec itself because warbler breaks w/o it
   s.files += %w[bundler.gemspec]
 
-  s.bindir        = "exe"
-  s.executables   = %w[bundle bundler]
+  # s.bindir        = "exe"
+  # s.executables   = %w[bundle bundler]
   s.require_paths = ["lib"]
 end
diff --git a/libexec/bundle b/libexec/bundle
deleted file mode 100755
index aaf773745d..0000000000
--- a/libexec/bundle
+++ /dev/null
@@ -1,31 +0,0 @@
-#!/usr/bin/env ruby
-# frozen_string_literal: true
-
-# Exit cleanly from an early interrupt
-Signal.trap("INT") do
-  Bundler.ui.debug("\n#{caller.join("\n")}") if defined?(Bundler)
-  exit 1
-end
-
-require "bundler"
-# Check if an older version of bundler is installed
-$LOAD_PATH.each do |path|
-  next unless path =~ %r{/bundler-0\.(\d+)} && $1.to_i < 9
-  err = String.new
-  err << "Looks like you have a version of bundler that's older than 0.9.\n"
-  err << "Please remove your old versions.\n"
-  err << "An easy way to do this is by running `gem cleanup bundler`."
-  abort(err)
-end
-
-require "bundler/friendly_errors"
-Bundler.with_friendly_errors do
-  require "bundler/cli"
-
-  # Allow any command to use --help flag to show help for that command
-  help_flags = %w[--help -h]
-  help_flag_used = ARGV.any? {|a| help_flags.include? a }
-  args = help_flag_used ? Bundler::CLI.reformatted_help_args(ARGV) : ARGV
-
-  Bundler::CLI.start(args, :debug => true)
-end
diff --git a/libexec/bundle_ruby b/libexec/bundle_ruby
deleted file mode 100755
index 2209c6195f..0000000000
--- a/libexec/bundle_ruby
+++ /dev/null
@@ -1,60 +0,0 @@
-#!/usr/bin/env ruby
-# frozen_string_literal: true
-
-require "bundler/shared_helpers"
-
-Bundler::SharedHelpers.major_deprecation(3, "the bundle_ruby executable has been removed in favor of `bundle platform --ruby`")
-
-Signal.trap("INT") { exit 1 }
-
-require "bundler/errors"
-require "bundler/ruby_version"
-require "bundler/ruby_dsl"
-
-module Bundler
-  class Dsl
-    include RubyDsl
-
-    attr_accessor :ruby_version
-
-    def initialize
-      @ruby_version = nil
-    end
-
-    def eval_gemfile(gemfile, contents = nil)
-      contents ||= File.open(gemfile, "rb", &:read)
-      instance_eval(contents, gemfile.to_s, 1)
-    rescue SyntaxError => e
-      bt = e.message.split("\n")[1..-1]
-      raise GemfileError, ["Gemfile syntax error:", *bt].join("\n")
-    rescue ScriptError, RegexpError, NameError, ArgumentError => e
-      e.backtrace[0] = "#{e.backtrace[0]}: #{e.message} (#{e.class})"
-      STDERR.puts e.backtrace.join("\n       ")
-      raise GemfileError, "There was an error in your Gemfile," \
-        " and Bundler cannot continue."
-    end
-
-    def source(source, options = {})
-    end
-
-    def gem(name, *args)
-    end
-
-    def group(*args)
-    end
-  end
-end
-
-dsl = Bundler::Dsl.new
-begin
-  dsl.eval_gemfile(Bundler::SharedHelpers.default_gemfile)
-  ruby_version = dsl.ruby_version
-  if ruby_version
-    puts ruby_version
-  else
-    puts "No ruby version specified"
-  end
-rescue Bundler::GemfileError => e
-  puts e.message
-  exit(-1)
-end
diff --git a/libexec/bundler b/libexec/bundler
deleted file mode 100755
index d9131fe834..0000000000
--- a/libexec/bundler
+++ /dev/null
@@ -1,4 +0,0 @@
-#!/usr/bin/env ruby
-# frozen_string_literal: true
-
-load File.expand_path("../bundle", __FILE__)
diff --git a/libexec/irb b/libexec/irb
deleted file mode 100755
index c64ee85fbd..0000000000
--- a/libexec/irb
+++ /dev/null
@@ -1,11 +0,0 @@
-#!/usr/bin/env ruby
-#
-#   irb.rb - interactive ruby
-#   	$Release Version: 0.9.6 $
-#   	$Revision$
-#   	by Keiju ISHITSUKA(keiju@ruby-lang.org)
-#
-
-require "irb"
-
-IRB.start(__FILE__)
diff --git a/libexec/rdoc b/libexec/rdoc
deleted file mode 100755
index aaa23292df..0000000000
--- a/libexec/rdoc
+++ /dev/null
@@ -1,44 +0,0 @@
-#!/usr/bin/env ruby
-#
-#  RDoc: Documentation tool for source code
-#        (see lib/rdoc/rdoc.rb for more information)
-#
-#  Copyright (c) 2003 Dave Thomas
-#  Released under the same terms as Ruby
-
-begin
-  gem 'rdoc'
-rescue NameError => e # --disable-gems
-  raise unless e.name == :gem
-rescue Gem::LoadError
-end
-
-require 'rdoc/rdoc'
-
-begin
-  r = RDoc::RDoc.new
-  r.document ARGV
-rescue Errno::ENOSPC
-  $stderr.puts 'Ran out of space creating documentation'
-  $stderr.puts
-  $stderr.puts 'Please free up some space and try again'
-rescue SystemExit
-  raise
-rescue Exception => e
-  if $DEBUG_RDOC then
-    $stderr.puts e.message
-    $stderr.puts "#{e.backtrace.join "\n\t"}"
-    $stderr.puts
-  elsif Interrupt === e then
-    $stderr.puts
-    $stderr.puts 'Interrupted'
-  else
-    $stderr.puts "uh-oh! RDoc had a problem:"
-    $stderr.puts e.message
-    $stderr.puts
-    $stderr.puts "run with --debug for full backtrace"
-  end
-
-  exit 1
-end
-
diff --git a/libexec/ri b/libexec/ri
deleted file mode 100755
index 7fbed0c099..0000000000
--- a/libexec/ri
+++ /dev/null
@@ -1,12 +0,0 @@
-#!/usr/bin/env ruby
-
-begin
-  gem 'rdoc'
-rescue NameError => e # --disable-gems
-  raise unless e.name == :gem
-rescue Gem::LoadError
-end
-
-require 'rdoc/ri/driver'
-
-RDoc::RI::Driver.run ARGV
diff --git a/spec/bundler/spec_helper.rb b/spec/bundler/spec_helper.rb
index c7614e1c43..ff009e773f 100644
--- a/spec/bundler/spec_helper.rb
+++ b/spec/bundler/spec_helper.rb
@@ -131,12 +131,6 @@ def self.ruby=(ruby)
     Gem.ruby = orig_ruby if ENV["BUNDLE_RUBY"]
   end
 
-  config.before :suite do
-    if ENV["BUNDLE_RUBY"]
-      FileUtils.cp_r Spec::Path.bindir, File.join(Spec::Path.root, "lib", "exe")
-    end
-  end
-
   config.before :all do
     build_repo1
   end
@@ -161,10 +155,4 @@ def self.ruby=(ruby)
     Dir.chdir(original_wd)
     ENV.replace(original_env)
   end
-
-  config.after :suite do
-    if ENV["BUNDLE_RUBY"]
-      FileUtils.rm_rf File.join(Spec::Path.root, "lib", "exe")
-    end
-  end
 end
diff --git a/spec/bundler/support/path.rb b/spec/bundler/support/path.rb
index 03a96893f3..97153226bd 100644
--- a/spec/bundler/support/path.rb
+++ b/spec/bundler/support/path.rb
@@ -13,7 +13,7 @@ def gemspec
     end
 
     def bindir
-      @bindir ||= root.join(ruby_core? ? "libexec" : "exe")
+      @bindir ||= root.join(ruby_core? ? "bin" : "exe")
     end
 
     def spec_dir
diff --git a/tool/rbinstall.rb b/tool/rbinstall.rb
index 13a897cdc5..b448e62d07 100755
--- a/tool/rbinstall.rb
+++ b/tool/rbinstall.rb
@@ -781,7 +781,7 @@ def install_default_gem(dir, srcdir)
       makedirs(bin_dir)
 
       gemspec.executables.map {|exec|
-        $script_installer.install(File.join(srcdir, 'libexec', exec),
+        $script_installer.install(File.join(srcdir, 'bin', exec),
                                   File.join(bin_dir, exec))
       }
     end
